src/main/java/io/cdap/plugin/successfactors/source/config/SuccessFactorsPluginConfig.java

```diff
@@ -76,38 +95,60 @@ public class SuccessFactorsPluginConfig extends PluginConfig {
 
   @Nullable
   @Macro
-  @Description("Fields to be preserved in the extracted data e.g.: Name,Gender,State/City")
+  @Description("Fields to be preserved in the extracted data. e.g.: Category, Price, Name, Address. If the field is " +
+    "left blank, then all the non-navigation fields will be preserved in the extracted data.\n" +
+    "All the fields must be comma (,) separated.\n")
   private final String selectOption;
 
   @Nullable
   @Macro
-  @Description("List of navigation fields to be expanded in the extracted output data e.g.: State/City")
+  @Description("List of navigation fields to be expanded in the extracted output data. For example: customManager. " +
+    "If an entity has hierarchical records, the source outputs a record for each row in the entity it reads, with " +
+    "each record containing an additional field that holds the value from the navigational property specified in " +
+    "the Expand Fields.")
   private final String expandOption;
-  
+
   /**
    * Basic parameters.
    */
   @Name(REFERENCE_NAME)
   @Description(REFERENCE_NAME_DESCRIPTION)
   public String referenceName;
 
+  @Name(NAME_SCHEMA)
+  @Macro
+  @Nullable
+  @Description("The schema of the table to read.")
+  private String schema;
+
+  @Name(PAGINATION_TYPE)
+  @Macro
+  @Description("The type of pagination to be used. Server-side Pagination uses snapshot-based pagination. " +
+    "If snapshot-based pagination is attempted on an entity that doesn’t support the feature, the server " +
+    "automatically forces client-side pagination on the query. Default is Server-side Pagination.")
```

Comment:
Can you point me to where we are handling this case in the PR (server-side pagination is specified in config, but we end up using client-side pagination)?

Reply:
It is implicitly handled by the server. There is no handling in the code.

Reply:
Plugin code handles server side vs client side pagination differently. In the case of server side pagination there's only 1 split and we rely on a nexturl being returned by the server.
In the case of client side pagination, there are multiple splits, and we make API calls specifying number of rows to skip, fetch.
Plugin code determines whether to use client-side or server-side processing logic depending on the plugin config.
It's not clear to me how the plugin handles a client-side pagination response when it's making a server-side pagination request, expects a response containing the next url to process. Can you please clarify how that works?

Reply:
From [SAP Docs](https://help.sap.com/docs/SAP_SUCCESSFACTORS_PLATFORM/d599f15995d348a1b45ba5603e2aba9b/7af42e7079c3481688de975b38867c41.html):
> Not all entities support snapshot-based pagination. If you attempt snapshot-based pagination on an entity that doesn’t support the feature, the server automatically forces client offset pagination on the query. You can find out which type of pagination is used in the following custom headers of the query response:
If snapshot-based pagination is successful: X-SF-Paging: snapshot
If client offset pagination is successful as a fallback: X-SF-Paging: offset

During our testing, we observed that the following three entities meet the criteria of client offset pagination as a fallback

1. BadgeTemplates
2. UserBadges
3. EPCustomBackgroundPortlet

The `$count` API for all 3 entities returns the count as -1 
So, no records are transferred for these entities with client-side pagination, warning that no records were found in this entity. The same has been added to the user guide.
But it was observed that all the records are transferred with server-side pagination.

The API Response for these entities does not return the `__next` pointer when queried with: `paging=snapshot` query param. For one of the entities, BadgeTemplates when it had 1207 records, the pipeline could transfer all the records. All the records even greater than MAX_ALLOWED_BATCH_SIZE are fetched in a single call.

So, to conclude, paging is not supported in these entities. The plugin doesn't handle it. The server is implicitly handling it.